main.go

```diff
@@ -11,165 +13,126 @@ import (
-			os.Exit(1)
+		litter.Config.HideZeroValues = true
+		litter.Config.HidePrivateFields = true
+		// Save the stages to a file for debugging and future use
+		if config.DefaultConfig.Stage == "all" {
+			_ = os.WriteFile("/etc/kairos/kairos-init-all-stage.yaml", []byte(runStages.ToString()), 0644)
+		} else {
+			_ = os.WriteFile(fmt.Sprintf("/etc/kairos/kairos-init-%s-stage.yaml", config.DefaultConfig.Stage), []byte(runStages.ToString()), 0644)
 		}
-	}
 
-	if err != nil {
-		logger.Error(err)
-		os.Exit(1)
-	}
+		return nil
+	},
+}
 
-	litter.Config.HideZeroValues = true
-	litter.Config.HidePrivateFields = true
-	// I would say lets save the stages to a file for debugging and future use
-	// we don't fail if we cant write the file
-	if config.DefaultConfig.Stage == "all" {
-		_ = os.WriteFile("/etc/kairos/kairos-init-all-stage.yaml", []byte(runStages.ToString()), 0644)
-	} else {
-		_ = os.WriteFile(fmt.Sprintf("/etc/kairos/kairos-init-%s-stage.yaml", config.DefaultConfig.Stage), []byte(runStages.ToString()), 0644)
-	}
+func init() {
+	// Global flags
+	rootCmd.Flags().StringVarP(&config.DefaultConfig.Level, "level", "l", "info", "set the log level")
+	rootCmd.Flags().StringVarP(&config.DefaultConfig.Stage, "stage", "s", "all", "set the stage to run")
+	rootCmd.Flags().StringVarP(&config.DefaultConfig.Model, "model", "m", "generic", "model to build for, like generic or rpi4")
+	rootCmd.Flags().StringVarP(&variant, "variant", "v", config.CoreVariant.String(), "variant to build (core or standard for k3s flavor)")
+	rootCmd.Flags().StringVarP(&ksProvider, "kubernetes-provider", "k", string(config.K3sProvider), "Kubernetes provider")
+	rootCmd.Flags().StringVar(&config.DefaultConfig.KubernetesVersion, "k8sversion", "latest", "Kubernetes version for provider")
+	rootCmd.Flags().StringVarP(&config.DefaultConfig.Registry, "registry", "r", "quay.io/kairos", "registry and org where the image is gonna be pushed. This is mainly used on upgrades to search for available images to upgrade to")
+	rootCmd.Flags().StringVarP(&trusted, "trusted", "t", "false", "init the system for Trusted Boot, changes bootloader to systemd")
+	rootCmd.Flags().StringVarP(&config.DefaultConfig.FrameworkVersion, "framework", "f", values.GetFrameworkVersion(), "set the framework version to use")
+	rootCmd.Flags().BoolVar(&validate, "validate", false, "validate the running os to see if it all the pieces are in place")
+	rootCmd.Flags().BoolVar(&config.DefaultConfig.Fips, "fips", false, "use fips framework. For FIPS 140-2 compliance images")
+	rootCmd.Flags().StringVar(&version, "version", "", "set a version number to use for the generated system. Its used to identify this system for upgrades and such. Required.")
+	rootCmd.Flags().BoolVar(&config.DefaultConfig.Extensions, "stage-extensions", false, "enable stage extensions mode")
+
+	// Mark required flags
+	rootCmd.MarkFlagRequired("version")
```

Comment:
arent we also adding the registry as required?

Reply:
true, I was just solving the library migration problem, but I'll do that here since we are at it.

Reply:
done

Reply:
Every call to `kairos-init` needs to be updated now, to pass the registry... I expect some pipelines to fail.